src/Umbraco.Web.UI.Client/src/views/propertyeditors/blocklist/blocklist.component.html

```diff
@@ -0,0 +1,88 @@
+<div class="umb-block-list">
+
+    <umb-load-indicator style="margin-bottom:20px;" ng-if="vm.loading"></umb-load-indicator>
+
+    <div class="umb-block-list__wrapper" ng-style="vm.listWrapperStyles">
+
+        <div ui-sortable="vm.sortableOptions" ng-model="vm.layout" ng-if="vm.loading !== true">
+
+            <div ng-repeat="layout in vm.layout track by layout.$block.key">
+
+                <button
+                    type="button"
+                    class="btn-reset umb-block-list__block--create-button"
+                    ng-click="vm.showCreateDialog($index, $event)"
+                    ng-controller="Umbraco.PropertyEditors.BlockListPropertyEditor.CreateButtonController as inlineCreateButtonCtrl"
```

Comment:
Components are statically defined, this is very specific to this usage and is made as a controller causes the functionality wasn't possible to make with inline JS and since its part of the repeat, it would not be easy to make it as part of the component controller. So this was a good way to solve the need without making a new component that others might start using.

Reply:
OK sounds very edge case then